gammapy/modeling/models/cube.py

```diff
@@ -607,6 +607,7 @@ def __init__(self, spectral_model=None, dataset_name=None):
         if not spectral_model.is_norm_spectral_model:
             raise ValueError("A norm spectral model is required.")
 
+        self._spatial_model = None
```

Comment:
why here do we not have:
self._spatial_model = spatial_model?
Like for the spectral_model....